source/development/core-dumps-generation.rst

```diff
@@ -0,0 +1,152 @@
+.. Copyright (C) 2022 Wazuh, Inc.
+
+.. meta::
+  :description: Core dump generation and analysis
+  
+.. _dev-core-dumps-generation:
+
+
+Core dump generation and analysis
+=================================
+
+A core dump is a snapshot of the process state generated when a process crashes or stops abnormally.
+It lets you view part of the state the process was in when it crashed, this includes the stack trace, variable and register values, etc.
+
+This article describes how to enable core dump generation and how to use a debugger for post-mortem crash analysis.
+
+Some operating systems perform a core dump upon a process crash by default, others do not. This article covers how to
+enable and disable core dump generation on the supported platforms [1]_.
+
+.. [1] As the time of writing, the Wazuh compilation outputs separate debugging symbols only for Windows, macOS, and Linux.
+
+macOS
+*****
+
+Enabling core dump generation
+-----------------------------
+
+To enable core dumps in macOS, you need to change the resource limits of a process using `ulimit`.  In this case, the
+resource limit you need to change is the "cores" one, which represents the maximal size of core dumps our process is
+allowed to generate.
+
+To enable core dump generation for Wazuh, the service needs to be relaunched from a shell, this is because resource
+limits are propagated to child processes, so if we set the core limit to a certain value on a shell, then all
+processes spawned by it (for example by invoking a command) will inherit the resource limits. Normally, all Wazuh
+processes have launchd as their parent process, which will only propagate the default system-wide resource limits.
+
+Run the following commands as root. First, stop the service::
+
+  # /Library/Ossec/bin/wazuh-control stop
+
+Then, use ``ulimit`` to set the max core dump size to unlimited::
+
+  # ulimit -c unlimited
+
+Keep in mind that doing this will result in every command run from within this shell session will have no limit on the
+size of core dumps.
+
+There is a system setting that defines the path where core dumps will be stored. It is usually ``/cores/`` but we can
+check this with::
+
+  # sysctl kern.corefile
+
+The default output of this command will be "/cores/core.%P" which means core dumps will comply to this formatting rule.
+We will assume this is the case for the rest of this article.
+
+Next, make sure the ``/cores`` folder exists::
```

Comment:
Maybe add folder's privileges aclaration?

Reply:
Yeah, good point. adding it

Reply:
Added `chmod 755 /cores` to the commands for creating the directory.